src/main/java/jenkins/plugins/logstash/persistence/ElasticSearchDao.java

```diff
@@ -0,0 +1,135 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.plugins.logstash.persistence;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.client.utils.URIBuilder;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import java.io.PrintStream;
+
+/**
+ * Elastic Search Data Access Object.
+ *
+ * @author Liam Newman
+ * @since 1.0.5
+ */
+public class ElasticSearchDao extends AbstractLogstashIndexerDao {
+  HttpClientBuilder clientBuilder;
+  URI uri;
+  String auth;
+
+
+  ElasticSearchDao() { /* Required by IndexerDaoFactory */ }
+
+  // Constructor for unit testing
+  ElasticSearchDao(String host, int port, String key, String username, String password) {
+    init(host, port, key, username, password);
+  }
+  final void init(String host, int port, String key, String username, String password) {
+    super.init(host, port, key, username, password);
+
+    if (StringUtils.isBlank(key)) {
+      throw new IllegalArgumentException("elastic index name is required");
+    }
+
+    try {
+      uri = new URIBuilder(host)
+        .setPort(port)
+        // Normalizer will remove extra starting slashes, but missing slash will cause annoying failures
+        .setPath("/" + key)
+        .build();
+    } catch (URISyntaxException e) {
+      throw new IllegalArgumentException("Could not create uri", e);
+    }
+
+    auth = null;
+    if (StringUtils.isNotBlank(username)) {
+      auth = Base64.encodeBase64String((username + ":" + StringUtils.defaultString(password)).getBytes());
+    }
+
+    clientBuilder = HttpClientBuilder.create();
+  }
+
+  @Override
+  public long push(String data, PrintStream logger) {
+    CloseableHttpClient httpClient = null;
+    CloseableHttpResponse response = null;
+    try {
+      httpClient = clientBuilder.build();
+      HttpPost postRequest = new HttpPost(uri);
+      StringEntity input = new StringEntity(data, "UTF-8");
+      input.setContentType("application/json");
```

Comment:
Minor nitpick: there's constants defined for these strings in java.nio.charset.StandardCharsets and org.apache.http.entity.ContentType:

`StandardCharsets.UTF_8`
`ContentType.APPLICATION_JSON`